plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -873,13 +872,27 @@ private static Set<String> getOutputFilesLocations(Set<String> writtenFiles)
                 .collect(toImmutableSet());
     }
 
+    private static Set<String> getOutputFilesFileNames(Set<String> writtenFiles)
+    {
+        return writtenFiles.stream()
+                .map(IcebergMetadata::getFileName)
+                .collect(toImmutableSet());
+    }
+
     private static String getLocation(String path)
     {
         Matcher matcher = PATH_PATTERN.matcher(path);
         verify(matcher.matches(), "path %s does not match pattern", path);
         return matcher.group(1);
     }
 
+    private static String getFileName(String path)
```

Comment:
Other places in this file we've been using `IcebergUtil#fileName` which uses substring instead of regex